.github/docs/qemu.md

```diff
@@ -0,0 +1,93 @@
+# Local image testing with QEMU / libvirt
+
+To create local testing clusters using QEMU, some prerequisites have to be met:
+
+- [qcow2 constellation image](/image/README.md)
+- [libvirt setup](#setup-libvirt)
+- [qemu-metadata-api container image](/hack/qemu-metadata-api/README.md)
+
+## Setup libvirt
+
+<details>
+<summary>Ubuntu</summary>
+
+### Install required packages
+
+[General reference](https://ubuntu.com/server/docs/virtualization-libvirt)
+
+```shell-session
+sudo apt install qemu-kvm libvirt-daemon-system xsltproc
+sudo systemctl enable libvirtd
+sudo usermod -a -G libvirt $USER
+# reboot
+```
+
+### Setup emulated TPM
+
+Using a virtual TPM (vTPM) with QEMU only works if swtpm is version 0.7 or newer!
+Ubuntu 22.04 currently ships swtpm 0.6.3, so you need to install swtpm [from launchpad](https://launchpad.net/~stefanberger/+archive/ubuntu/swtpm-jammy/).
+
+1. Uninstall current version of swtpm (if installed)
+
+    ```shell-session
+    sudo apt remove swtpm swtpm-tools
+    ```
+
+2. Add ppa (this command shows the ppa for Ubuntu 22.04 jammy but others are available)
+
+    ```shell-session
+    sudo add-apt-repository ppa:stefanberger/swtpm-jammy
+    sudo apt update
+    ```
+
+3. Install swtpm
+
+    ```shell-session
+    sudo apt install swtpm swtpm-tools
+    ```
+
+4. Patch configuration under `/etc/swtpm_setup.conf`
+
+    ```shell-session
+    # Program invoked for creating certificates
+    create_certs_tool = /usr/bin/swtpm_localca
+    ```
+
+5. Patch ownership of `/var/lib/swtpm-localca`
+
+   ```shell-session
+   sudo chown -R swtpm:root /var/lib/swtpm-localca
+   ```
+
+</details>
+
+<details>
+<summary>Fedora</summary>
+
+```shell-session
+sudo dnf install -y dnf-plugins-core
+sudo dnf -y install qemu-kvm libvirt-daemon-config-network libvirt-daemon-kvm xsltproc swtpm
+sudo usermod -a -G libvirt $USER
+# reboot
+```
+
+</details>
+
+## Update libvirt settings
+
+Open `/etc/libvirt/qemu.conf` and change the following settings:
+
+```shell-session
+security_driver = "none"
```

Comment:
Not sure about Fedora/SELinux, but I think on Ubuntu you can get away with keeping AppArmor disabled but defining:
```
"/var/lib/libvirt/images" r,
"/var/lib/libvirt/images/**" rwk,
```
in `/etc/apparmor.d/local/abstractions/libvirt-qemu`.  At least that works on my machine™

Not sure if we want to suggest this to the user, though if it can cause issues on other distros.

Reply:
I like the idea of defining rules for only what is needed for AppArmor. Disabling the security_driver on Ubuntu seems to be what everyone else is doing:

- https://github.com/openshift/installer/blob/master/docs/dev/libvirt/README.md#configure-qemuconf
- https://github.com/dmacvicar/terraform-provider-libvirt/issues?q=security_driver

I think we can test your approach and see if it breaks :)

Reply:
I mean, it works (or worked :P) on my machine and many results about this issue mainly revolve around storage pools. The exemption should cover it for our usage (though technically breaking guest-to-guest if a guest can manipulate qemu?), but might not work for other libvirt applications. 

So that's why I can see others suggesting to completely disable it. Or people just choose the lazy way to fix this issue, that wouldn't surprise me either.

There might even be a better, stricter way to fix this to keep the isolation in-tact for our use-case. But my current testing setup is broken for some other reason 🙃